File: BaekJoon/solved/JS/1835 카드.js
let I = +(require('fs').readFileSync(0, 'utf-8') + '').trim();

let F=1000, B=999;
let L = new Array(2000).fill(0);

const push_front = (v) => {
    L[--F] = v;
}
const pop_front = () => {
    let R = L[F];
    L[F++] = 0;
    return R;
}
const pop_back = () => {
    let R = L[B];
    L[B--] = 0;
    return R;
}

function BOJ_1835()
{
	push_front(I--);
	while (I > 0)
	{
		push_front(I--);
		for (let i = 0; i <= I; i++)
			push_front(L[B]), pop_back();
	}
    S = ""
	while (F <= B)
		S += `${L[F]} `, pop_front();
    console.log(S);
}

BOJ_1835();

File: BaekJoon/solved/C++/1835 카드.cpp
#include <iostream>
#include <deque>
using namespace std;

void BOJ_1835()
{
	int N;
	cin >> N;
	deque<int> D;
	D.push_front(N--);
	while (N > 0)
	{
		D.push_front(N--);
		for (int i = 0; i <= N; i++)
			D.push_front(D.back()), D.pop_back();
	}
	while (!D.empty())
		cout << D.front() << " ", D.pop_front();
}

int main()
{
	ios::sync_with_stdio(0), cin.tie(0);
	BOJ_1835();
	return (0);
}